File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        vector<int> ans;
        ans.push_back(nums[0]);
        for (int i = 1; i < n; i++) {
            if (nums[i] != ans.back())
                ans.push_back(nums[i]);
        }
        for (int i = 0; i < ans.size(); i++) {
            nums[i] = ans[i];
        }
        return ans.size();
    }
};

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n=nums.length;
        int prev=0;
        int next=1;
        int count=0;
        while(next<n){

            if(nums[prev]!=nums[next]){
                prev++;
                int temp=nums[prev];
                nums[prev]=nums[next];
                nums[next]=temp;
                count++;
            }
            next++;
        }
        return count+1;
    }
}